ociclient/ociclient.go

```diff
@@ -0,0 +1,153 @@
+/*
+ *   Copyright (c) 2024 Oracle Corporation
+ *   All rights reserved.
+ *   SPDX-License-Identifier: BSD-3-Clause
+ */
+
+package ociclient
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"intel/kbs/v1/model"
+	"regexp"
+	"time"
+
+	"github.com/oracle/oci-go-sdk/v65/common"
+	"github.com/oracle/oci-go-sdk/v65/secrets"
+	"github.com/oracle/oci-go-sdk/v65/vault"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+)
+
+type OCIClient interface {
+	InitializeClient() error
+	CreateKey(keyAttrib *model.KeyAttributes) error
+	DeleteKey(string) error
+	GetKey(string, int64) ([]byte, error)
+}
+
+type ociClient struct {
+	sc *(secrets.SecretsClient)
+	vc *(vault.VaultsClient)
+}
+
+func NewOCIClient() OCIClient {
+	return &ociClient{}
+}
+
+func (oc *ociClient) InitializeClient() error {
+	secretsClient, err := secrets.NewSecretsClientWithConfigurationProvider(common.DefaultConfigProvider())
+	if err != nil {
+		return errors.Wrap(err, "ociclient/ociclient:InitializeClient() Failed to initialize OCI secrets client")
+	}
+
+	vaultsClient, err := vault.NewVaultsClientWithConfigurationProvider(common.DefaultConfigProvider())
+	if err != nil {
+		return errors.Wrap(err, "ociclient/ociclient:InitializeClient() Failed to initialize OCI vaults client")
+	}
+
+	oc.sc = &secretsClient
+	oc.vc = &vaultsClient
+
+	log.Info("ociclient/ociclient:InitializeClient() OCI client initialized")
+
+	return nil
+}
+
+func (oc *ociClient) CreateKey(keyAttributes *model.KeyAttributes) error {
+	// Create request and dependent object(s).
+	req := vault.CreateSecretRequest{
+		CreateSecretDetails: vault.CreateSecretDetails{
+			// Required
+			CompartmentId: common.String(keyAttributes.OciCompartmentId),
```

Comment:
just a thought, would it be better to pass only the required arguments to this function instead of keyAttributes struct, just to keep the client independent of keyAttributes implementation.

Reply:
I went back and forth on that one. In the end, I again decided to model it after the Vault implementation. vaultclient.go:CreateKey() also takes a KeyAttributes struct.

I'm happy to change it to take the required arguments, though. I can also (in a separate pull request) also similarly update the vaultclient code.

Reply:
checked vaultclient implementation, its bit different in the sense that it sends entire KeyAttributes struct to backend. I would let @hmgowda comment on updating vaultclient code.